pool.go

```diff
@@ -0,0 +1,53 @@
+package shredis
+
+import (
+	"errors"
+)
+
+type pool struct {
+	addr      string
+	conns     chan *conn
+	onConnect []*Cmd
+}
+
+func newPool(addr string, size int, onConnect []*Cmd) *pool {
+	cs := make(chan *conn, size)
+	for i := 0; i < size; i++ {
+		cs <- nil
+	}
+	return &pool{
+		addr:      addr,
+		conns:     cs,
+		onConnect: onConnect,
+	}
+}
+
+// Close blocks until all connections are done.
+func (p *pool) Close() {
+	close(p.conns)
+	for i := 0; i < cap(p.conns); i++ {
+		if c := <-p.conns; c != nil {
+			c.c.Close()
+		}
+	}
+}
```

Comment:
The way I see it, `Exec` after `Close` errors, `Exec` that finished before `Close` is fine, and `Exec` that starts before `Close` and finishes after panics. Might as well insist that the user waits for their execs to finish before closing, but then blocking makes no sense anymore...